TP-Sampler/src/app.js

```diff
@@ -666,97 +666,133 @@ function setup() {
-    });
-  }
+      function createAttrOptions (selectEl) {
+        const attrOptions = Object.keys(attrMap).filter((attrName) => attrMap[attrName].length > 0);
+        attrOptions.forEach((option, i) => {
+          const outputOption = document.createElement("option");
+          outputOption.value = option;
+          outputOption.textContent = option;
+          if (i === 0) {
+            outputOption.selected = true;
+          }
+          selectEl.appendChild(outputOption);
+        });
+      };
+
+      function createValueOptions (selectEl, selectedAttribute) {
+        const attrValues = attrMap[selectedAttribute];
+        if (attrValues.length) {
+          const options = attrValues.map((v) => {return {value: v, text: v}});
+          options.forEach((option, i) => {
+            const valueOption = document.createElement("option");
+            valueOption.value = option.value;
+            valueOption.textContent = option.text;
+            if (i === 0) {
+              valueOption.selected = true;
+            }
+            selectEl.appendChild(valueOption);
+          });
+          return options;
+        }
+      };
 
-  function createOperatorOptions (selectEl) {
-    function isStringNumber(str) {
-      return !isNaN(parseFloat(str)) && isFinite(str);
-    }
-    const anyVariableIsNumber = variables.some((v) => isStringNumber(v));
-    if (anyVariableIsNumber) {
-      const options = ["=", "≠", "<", ">", "≤", "≥"]
-      options.forEach((option, i) => {
-        const operatorOpt = document.createElement("option");
-        operatorOpt.value = option;
-        operatorOpt.textContent = option;
-        if (i === 0) {
-          operatorOpt.selected = true;
+      function createOperatorOptions (selectEl, values) {
+        function isStringNumber(str) {
+          return !isNaN(parseFloat(str)) && isFinite(str);
         }
-        selectEl.appendChild(operatorOpt);
-      })
-    } else {
-      const equalsOpt = document.createElement("option");
-      equalsOpt.value = "=";
-      equalsOpt.textContent = "=";
-      selectEl.appendChild(equalsOpt);
-      const notEqualsOpt = document.createElement("option");
-      notEqualsOpt.value = "≠";
-      notEqualsOpt.textContent = "≠";
-      selectEl.appendChild(notEqualsOpt);
-    }
-  }
+        const anyVariableIsNumber = values.some((v) => isStringNumber(v.value));
+        if (anyVariableIsNumber) {
+          const options = ["=", "≠", "<", ">", "≤", "≥"]
+          options.forEach((option, i) => {
+            const operatorOpt = document.createElement("option");
+            operatorOpt.value = option;
+            operatorOpt.textContent = option;
+            if (i === 0) {
+              operatorOpt.selected = true;
+            }
+            selectEl.appendChild(operatorOpt);
+          })
+        } else {
+          const equalsOpt = document.createElement("option");
+          equalsOpt.value = "=";
+          equalsOpt.textContent = "=";
```

Comment:
Should `equalsOpt.selected = true` be set to match the other cases?

Reply:
I think it's OK as is, it seems to work as intended.